src/routes/guards/withdrawGuard.tsx

```diff
@@ -0,0 +1,22 @@
+import { Navigate, Outlet, useLocation } from 'react-router-dom';
+import { ROUTES_CONFIG } from '@/routes/routesConfig';
+import { useWithdrawStore } from '@/common/stores/withdraw';
+
+const WithdrawGuard = () => {
+  const { search } = useLocation();
+  const step = new URLSearchParams(search).get('step');
+  const isValidatedWithdraw = useWithdrawStore((s) => s.isValidatedWithdraw);
+  const isWithdrawCompleted = useWithdrawStore((s) => s.isWithdrawCompleted);
+
+  if (!isValidatedWithdraw) {
+    return <Navigate to={ROUTES_CONFIG.mypage.path} replace />;
+  }
+
+  if (step === '3' && !isWithdrawCompleted) {
+    return <Navigate to={ROUTES_CONFIG.withdraw.path} replace />;
+  }
```

Comment:
물론 의미상으로 바로 알 수 있다면 길이와 상관없이 조건식에 조건을 포함시켜도 괜찮다고 생각하지만,
해당 `step === '3'`이라는 코드가 뭘 의미하는지 바로 보고는 이해하기가 쉽지 않아서 이를 따로 분리해서 이름을 붙여주면 어떨까요??

```suggestion
const COMPLETION_STEP_VALUE = '3';
const isOnCompletionStep = currentStep === COMPLETION_STEP_VALUE;

// ...

if (isOnCompletionStep && !isWithdrawCompleted) {
    return <Navigate to={ROUTES_CONFIG.withdraw.path} replace />;
}
```